src/common/components/video-upload-threespeak/index.tsx

```diff
@@ -0,0 +1,291 @@
+import React, { useState, ChangeEvent, useEffect, useRef } from "react";
+import { videoSvg, uploadSvgV } from "../../img/svg";
+import { Button, Modal } from "react-bootstrap";
+import { _t } from "../../i18n";
+import "./index.scss";
+import { getAllVideoStatuses, uploadVideoInfo } from "../../api/threespeak";
+import * as tus from "tus-js-client";
+import VideoGallery from "../video-gallery";
+import useMount from "react-use/lib/useMount";
+import { success } from "../feedback";
+import { ActiveUser } from "../../store/active-user/types";
+import { Global } from "../../store/global/types";
+
+interface Props {
+  global: Global;
+  activeUser: ActiveUser | null;
+  insertText: (before: string, after?: string) => void;
+  setVideoEncoderBeneficiary?: (video: any) => void;
+  toggleNsfwC?: () => void;
+}
+
+export const VideoUpload = (props: Props) => {
+  const { activeUser, global, insertText, setVideoEncoderBeneficiary, toggleNsfwC } = props;
+
+  const tusEndPoint = "https://uploads.3speak.tv/files/";
```

Comment:
@Adesojisouljay Would it be possible to move these endpoint and upload functions inside api/threespeak.ts file instead? Also please check that all helpers related to vide upload to be in that file, not in components. Otherwise we will struggle later to find which endpoint is down or needs updating.

Reply:
Okay, I'll check that also.
Thanks